arangod/RestHandler/RestSystemReportHandler.cpp

```diff
@@ -48,14 +49,17 @@ RestSystemReportHandler::RestSystemReportHandler(
   application_features::ApplicationServer& server, GeneralRequest* request,
   GeneralResponse* response) : RestBaseHandler(server, request, response) {}
 
-std::string executeOSCmd(std::string const& cmd) {
-  static std::string const tmpFile("/tmp/arango-inspect.tmp");
-  static std::string const pipe(" | tee ");
-  static std::string const terminate(" > /dev/null");
-  std::system ((cmd + pipe + tmpFile + terminate).c_str());
-  std::ifstream tmpStream(tmpFile.c_str());
-  return std::string(
-    std::istreambuf_iterator<char>(tmpStream), std::istreambuf_iterator<char>());
+std::string exec(std::string const& cmd) {
+  std::array<char, 128> buffer;
+  std::string result;
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
```

Comment:
The man pages for my system say that `popen` returns a `FILE*`:
```
FILE *popen(const char *command, const char *type);
```
However, they also say that `pipe` expects an `int[2]` as its input parameter, but no `FILE*`:
```
int pipe(int pipefd[2]);
```
How is this supposed to work?

Reply:
`pipe` is a variable name?

Reply:
A small reformulation could make that more clear:
```suggestion
  auto pipe = std::make_unique<FILE, decltype(&pclose)>(popen(cmd.c_str(), "r"), pclose);
```